pkg/virt-controller/watch/migration.go

```diff
@@ -406,7 +413,10 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 	}
 
 	// Remove the finalizer and conditions if the migration has already completed
-	if migration.IsFinal() {
+	if migration.IsFinal() || c.isMigrationAbortedBeforeHandoff(getMigrationKey(migration)) {
```

Comment:
you don't need to combine this finalizer logic with the `abortBeforeHandoff` logic. We don't do that for any of the other failure cases.

This abort before handoff failure case is just another if statement in this chain. It doesn't need to be unique.

Reply:
The reasoning behind it is to being able to abort the migration within a single control loop.
Originally, I wanted to have another `if` statement before the long `if-else` starts, e.g. something like:
```go
if c.isMigrationAbortedBeforeHandoff(getMigrationKey(migration)) {
	c.failMigration(true, migrationCopy, vmi, pod)
}

if migration.IsFinal() {
...
} else if ...
```

The problem though is that `migration.IsFinal()` is directed to the cached migration object, while `failMigration` cannot act on it, only on the copied vmim.

The other alternative is to fail the migration in one of the `if-else` clauses then wait for another control loop to finalize the vmim.

@vladikr WDYT?